src/turbomind/models/llama/LlamaBatch.cc

```diff
@@ -1162,14 +1162,13 @@ void LlamaBatch<T>::OutputContextLogits(T*                                  cont
 
     if (context_logits_buf_ == nullptr) {
         NcclGuard guard(model_->tensor_para_, stream_, true);
-        context_logits_buf_ =
-            (float*)allocator_->malloc(sizeof(float) * model_->vocab_size_padded_ * max_context_token_num_);
-        const auto tp = model_->tensor_para_.world_size_;
+        context_logits_buf_ = (float*)allocator_->malloc(sizeof(float) * model_->vocab_size_padded_ * num_token);
+        const auto tp       = model_->tensor_para_.world_size_;
```

Comment:
In current implmetation, these buffers are only going to be allocated once. `num_token` is just for this single iteration.